File: Bisection method/Bisection.cpp
#include <iostream>
#include <math.h>

using namespace std;


//f(x) = cos(4x^4+x)
double funkcja_trygonometryczna(double x)
{
	double x4 = pow(x, 4);
    double wynik = cos(4*x4+x);
	return wynik;
} 

//f(x) = 2x^7 + 4x^3 + 2x + 1 
double funkcja_wielomianowa(double x)
{
    double x7 = pow(x, 7);
    double x3 = pow(x, 3);
    double wynik = 2*x7 + 4*x3 + 2*x + 1;
    return wynik;
}

//f(x) = ln(x^2 + 2x) 
double funkcja_logarytmiczna(double x)
{
    double x2 = pow(x, 2);
    double wynik = log(x2 + 2*x);
    return wynik;
}

int main() 
{    
    double a, b, a1, b1, dokladnosc, srodek, srodek1;
    double licznik = 0; 
    cout.precision(4); //ustawiam precyzje do czterech miejsc po przecinku
    cout.setf(ios::fixed);

    poczatek:
    cout<<endl<<"       METODA BISEKCJI"<<endl;
    cout<<"Podaj początek przedziału (a): "; cin >> a;
    cout<<"Podaj koniec przedziału (b): "; cin >> b;
    cout<<"Podaj dokładność obliczen (np. 0.01): "; cin >> dokladnosc;

    //Obliczam wartości funkcji na granicy przedziałow
	a1 = funkcja_wielomianowa(a);
	b1 = funkcja_wielomianowa(b);	
	
	if(a1 * b1 < 0)
    {
         for(;;)
        {
            srodek = (a + b) / 2;
            srodek1 = funkcja_wielomianowa(srodek);

            if(fabs(srodek1) < dokladnosc)
               break; 

            licznik++;
            if(a1 * srodek1 < 0)
            {
                b = srodek;
                b1 = srodek1;
            }
            else
            {
                a = srodek; 
                a1 = srodek1;
            }
        }
    }
    else
    {
        cout << "Brak miejsc zerowych w przedziale. Podaj nowy przedział"<<endl;
        goto poczatek;
    }

    cout << endl << "Ilość zrealizowanych krokow obliczeń: " << licznik <<endl;
    cout << endl << "                 WYNIK" << endl;
    cout << "Punkt rozwiązania rownania "<<srodek<<endl;
    cout << "Wartosc funkcji w punkcjie "<<srodek<<" wynosi: "<<funkcja_wielomianowa(srodek)<<endl<<endl;
    
    return 0;
}

File: Bisection method/Bisection.swift
import Foundation

//f(x) = cos(4x^4+x)
func tryg(_ x:Double) -> Double
{
    let x4:Double = pow(x, 4)
    let wynik:Double = cos(4*x4+x)
    return wynik
}

//f(x) = 2x^7 + 4x^3 + 2x + 1
func wielo(_ x:Double) -> Double
{
    let x7:Double = pow(x, 7)
    let x3:Double = pow(x, 3)
    let wynik:Double = 2*x7 + 4*x3 + 2*x + 1
    return wynik
}

//f(x) = ln(x^2 + 2x)
func logarytm(_ x:Double) -> Double
{
    let x2:Double = pow(x, 2)
    let wynik:Double = log(x2 + 2*x)
    return wynik
}


var srodek:Double = 0.0
var srodek1:Double = 0.0
var licznik:Int = 0

print("       METODA BISEKCJI")

print("\nPodaj początek przedziału (a):")
var a = Double(readLine()!)

print("\nPodaj koniec przedziału (b):")
var b = Double(readLine()!)

print("\nPodaj dokładność obliczen (np. 0.01):")
var dokladnosc = Double(readLine()!)


var a1:Double = wielo(a!)
var b1:Double = wielo(b!)

if (a1*b1)<0
{
    while(true)
    {
        srodek = (a! + b!) / 2
        srodek1 = wielo(srodek)

        if(fabs(srodek1) < dokladnosc!)
        {
           break
        }
        
        licznik += 1
        
        if (a1 * srodek1) < 0
        {
            b = srodek;
            b1 = srodek1;
        }
        else
        {
            a = srodek;
            a1 = srodek1;
        }
    }
}
else
{
    print("Brak miejsc zerowych w przedziale. Podaj nowy przedział")
}

print("\nIlość zrealizowanych krokow obliczeń: \(licznik)")
print("\n\n                 WYNIK")
print("Punkt rozwiązania rownania \(srodek)")
print("Wartosc funkcji w punkcjie \(srodek) wynosi \(wielo(srodek))")
